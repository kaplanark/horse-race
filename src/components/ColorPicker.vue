<script setup>
import { computed } from 'vue';
const props = defineProps({
	name: {
		type: String,
		default: ''
	},
	value: {
		type: String,
		default: '#000000'
	},
});
const emit = defineEmits(['update:value']);

const inputHandler = (e) => emit('update:value', e.target.value);

const color = computed(() => {
	return props.value || '#000000';
});

</script>

<template>
	<div class="color-picker" :id="props.name">
		<label for="color" class="color-picker__wrapper" :style="{ '--color-select': props.value }">
			<input type="color" class="color-picker__input" name="color" id="color" :value="color" @input="inputHandler">
			<div class="color-picker__content">{{ color }}</div>
		</label>
	</div>
</template>

<style lang="scss">
.color-picker {
	&__wrapper {
		display: flex;
		align-items: center;
		justify-content: space-between;
		padding: 8px 16px;
		border: 1px solid #ccc;
		border-radius: 4px;
		cursor: pointer;
	}

	&__input {
		border: 0;
		outline: 0;
		background-color: transparent;
		width: 24px;
		height: 24px;
	}

	&__content {
		display: flex;
		align-items: center;
		justify-content: space-between;
	}
}
</style>