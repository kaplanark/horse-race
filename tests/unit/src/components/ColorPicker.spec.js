import { describe, expect, test } from 'vitest';
import { mount } from '@vue/test-utils';
import ColorPicker from '@components/ColorPicker.vue';

describe('ColorPicker.vue', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(ColorPicker, {
			props: {
				name: 'color',
				value: '#000000',
			}
		});
	});

	afterEach(() => {
		wrapper.unmount();
	});

	test('is correctly rendered', () => expect(wrapper.html()).toMatchSnapshot());

	test('is correctly rendered with a value', () => {
		expect(wrapper.get('input').element.value).toBe('#000000');
		expect(wrapper.get('.color-picker__content').text()).toBe('#000000');
	});

	test('is correctly rendered with a name', () => {
		wrapper.setProps({ name: 'color' });
		expect(wrapper.get('input').element.name).toBe('color');
	});

	test('emits the correct event when the input is changed', async () => {
		const input = wrapper.get('input');
		await input.setValue('#ffffff');
		await wrapper.get('input').trigger('change');
		expect(wrapper.emitted('update:value')).toBeTruthy();
	});
});
