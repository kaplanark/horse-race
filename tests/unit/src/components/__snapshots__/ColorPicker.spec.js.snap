// Vitest Snapshot v1

exports[`ColorPicker.vue > is correctly rendered 1`] = `
"<div class=\\"color-picker\\" id=\\"color\\"><label for=\\"color\\" class=\\"color-picker__wrapper\\"><input type=\\"color\\" class=\\"color-picker__input\\" name=\\"color\\" id=\\"color\\">
    <div class=\\"color-picker__content\\">#000000</div>
  </label></div>"
`;
